### PR TITLE
Fix incorrect AuthAccessCallback and AuthCallback

### PR DESCRIPTION
**Pull request by frimuchkov**
https://github.com/verdaccio/verdaccio/blob/master/src/lib/auth.ts#L264
https://github.com/verdaccio/verdaccio/blob/master/src/lib/auth.ts#L114

**Type:** bug
**Scope:** typings

### CODE REVIEW

File: core/types/index.d.ts [hunk 1]
```diff
@@ -496,8 +498,11 @@ declare module '@verdaccio/types' {
     tag?: string;
   }
 
-  type AuthAccessCallback = (error: string | null, access: boolean) => void;
-  type AuthCallback = (error: string | null, groups: string[] | false) => void;
+  // FIXME: error should be export type `VerdaccioError = HttpError & { code: number };` instead of AuthError
+  // but this type is on @verdaccio/commons-api and cannot be used here yet (I don't know why)
```
Comment (juanpicado) [inline core/types/index.d.ts:502]: Yeah, that's something I regret now. This types and also commons api will be part of the `5.x` so we won't have this issue anymore eventually.

File: core/types/index.d.ts [hunk 2]
```diff
@@ -1,5 +1,7 @@
 import { PassThrough } from 'stream';
 
+import { HttpError } from 'http-errors';
```
Comment (juanpicado) [inline core/types/index.d.ts:3]: We cannot do this, `'http-errors'` is a dependency which is no defined in the project. I'd rather copy the type to this file

like 

```
    interface HttpError extends Error {
        status: number;
        statusCode: number;
        expose: boolean;
        headers?: {
            [key: string]: string;
        };
        [key: string]: any;
    }
```

and we can remove it when commons-api and types lives in the same repo in the future. It is a workaround but I guess is valid for now.

Comment (juanpicado) [reply core/types/index.d.ts:3]: Or just add `"@types/http-errors"` as denpendency

Comment (frimuchkov) [reply core/types/index.d.ts]: Sure
Sorry, I made these changes via web-editor

**Comment by frimuchkov** [comment]
It seems everything is ok

**Comment by juanpicado** [comment]
Great job @frimuchkov  !!

**Pull request merged**